hypothesis-python/docs/reference/strategies.rst

```diff
@@ -0,0 +1,230 @@
+
+Numeric
+-------
+
+.. seealso::
+
+  See separate sections for :ref:`Numpy strategies <hypothesis-numpy>`, :ref:`Pandas strategies <hypothesis-pandas>`, and :ref:`Array API strategies <array-api>`.
+
+.. autofunction:: hypothesis.strategies.integers
+.. autofunction:: hypothesis.strategies.floats
+.. autofunction:: hypothesis.strategies.complex_numbers
+.. autofunction:: hypothesis.strategies.decimals
+.. autofunction:: hypothesis.strategies.fractions
+
+Strings
+-------
+
+.. seealso::
+
+  The |st.uuids| and |st.ip_addresses| strategies generate instances of :mod:`UUID <python:uuid>` and :mod:`IPAddress <python:ipaddress>` respectively. You can generate corresponding string values by using |strategy.map|, such as ``st.uuids().map(str)``.
+
+.. autofunction:: hypothesis.strategies.text
+.. autofunction:: hypothesis.strategies.characters
+.. autofunction:: hypothesis.strategies.from_regex
+.. autofunction:: hypothesis.strategies.binary
+.. autofunction:: hypothesis.strategies.emails
+
+Collections
+-----------
+
+.. autofunction:: hypothesis.strategies.lists
+.. autofunction:: hypothesis.strategies.tuples
+.. autofunction:: hypothesis.strategies.sets
+.. autofunction:: hypothesis.strategies.frozensets
+.. autofunction:: hypothesis.strategies.dictionaries
+.. autofunction:: hypothesis.strategies.fixed_dictionaries
+.. autofunction:: hypothesis.strategies.iterables
+
+Datetime
+--------
+
+.. autofunction:: hypothesis.strategies.dates
+.. autofunction:: hypothesis.strategies.times
+.. autofunction:: hypothesis.strategies.datetimes
+.. autofunction:: hypothesis.strategies.timezones
+.. autofunction:: hypothesis.strategies.timezone_keys
+.. autofunction:: hypothesis.strategies.timedeltas
+
+Recursive
+---------
+
+.. autofunction:: hypothesis.strategies.recursive
+.. autofunction:: hypothesis.strategies.deferred
+
+Random
+------
+
+.. autofunction:: hypothesis.strategies.randoms
+.. autofunction:: hypothesis.strategies.random_module
+.. autofunction:: hypothesis.register_random
+
+.. _combinators:
+
+Combinators
+-----------
+
+.. autofunction:: hypothesis.strategies.one_of
+.. autofunction:: hypothesis.strategies.builds
+.. autofunction:: hypothesis.strategies.composite
+.. autofunction:: hypothesis.strategies.data
+
+Typing
+------
+
+.. autofunction:: hypothesis.strategies.from_type
+.. autofunction:: hypothesis.strategies.register_type_strategy
+
+Hypothesis
+----------
+
+.. autofunction:: hypothesis.strategies.runner
+.. autofunction:: hypothesis.strategies.shared
+
+Misc
+----
+
+.. autofunction:: hypothesis.strategies.functions
+.. autofunction:: hypothesis.strategies.slices
+.. autofunction:: hypothesis.strategies.uuids
+.. autofunction:: hypothesis.strategies.ip_addresses
+
+.. autofunction:: hypothesis.strategies.sampled_from
+.. autofunction:: hypothesis.strategies.permutations
+
+Provisional
+-----------
+
+.. automodule:: hypothesis.provisional
+  :members:
+  :exclude-members: DomainNameStrategy
```

Comment:
We've only got two strategies in here; I'd be inclined to put them under `Strings` above.  (and include provisional in our all-documented test)

Reply:
I'm on board with this. How strongly should we be `.. warning` these strategies? I've never actually understood why these are provisional, the interface for domains and urls seem stable enough to me.

Reply:
Not strongly imo, they're provisional more because of iirc unresolved design questions than real instability.